py_ecc/bn128/bn128_field_elements.py

```diff
@@ -39,52 +40,54 @@ def inv(a, n):
 # A class for field elements in FQ. Wrap a number in this class,
 # and it becomes a field element.
 class FQ(object):
```

Comment:
I'm wondering if this shouldn't inherit from `int` or `numbers.Number`.  I think that it might simplify the typing (maybe) since each of the methods argument types that are currently `Union[int, "FQ")` could be reduced to just be `int`.

Reply:
@pipermerriam that would involve refactoring the whole codebase, and maybe we can open a new issue once we are done with this?